### PR TITLE
ICMSLST-2487 All Countries and Overseas regions

### PR DESCRIPTION
**Pull request by marcuspp**
- Add countries on this [government list](https://www.gov.uk/government/publications/uk-trade-tariff-country-and-currency-codes/uk-trade-tariff-country-and-currency-codes) that aren't already on the platform but sets them to inactive
- Add overseas regions to all countries using [regions list](https://data.trade.gov.uk/datasets/240d5034-6a83-451b-8307-5755672f881b)
- Add continent column to issued certificates report
- Refactors country test data to use django fixtures
- Changed view to "Add Country" to activate an inactive country. 
- All commission codes / hmrc codes now preset and are not editable

To test
- `git checkout ICMSLST-2487`
- `./scripts/reset-local-docker-db`
 - In a shell `Country.objects.filter(is_active=False).delete()`
 - In a shell `Country.objects.update(overseas_region=None)`
 - `python manage.py add_v2_reference_data`
 - `python manage.py check_data_counts` - Command will fail as cannot talk to v1 but will print `PASS` for the countries checks prior to the error

### CODE REVIEW

File: web/migrations/0001_initial.py [hunk 1]
```diff
@@ -23,8 +23,8 @@ class Migration(migrations.Migration):
     initial = True
 
     dependencies = [
-        ("auth", "0012_alter_user_first_name_max_length"),
         ("contenttypes", "0002_remove_content_type_name"),
+        ("auth", "0012_alter_user_first_name_max_length"),
```
Comment (chris-pettinga) [inline web/migrations/0001_initial.py:27]: interesting that the order of these keeps on switching around whenever anyone regenerates the migration file

Comment (MattHolmes123) [reply web/migrations/0001_initial.py:27]: Yer it kind of blows my mind that it's nondeterministic

File: web/templates/web/domains/country/add.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    {% include "web/domains/country/groups-tooltip.html" %}
+  {% endset %}
+  {{fields.field(form.name, tooltip=tooltip)}}
+{% endblock %}
+
+{% block actions %}
+<li>
+  <button id="btn-submit" type="submit" class="primary-button button">Add Country</button>
+</li>
+{% endblock %}
```
Comment (MattHolmes123) [inline web/templates/web/domains/country/add.html:19]: Missing newline

File: web/domains/country/views.py [hunk 3]
```diff
@@ -59,14 +59,18 @@ class CountryEditView(ModelUpdateView):
...
+    template_name = "web/domains/country/add.html"
     form_class = CountryCreateForm
-    success_url = reverse_lazy("country:list")
-    cancel_url = success_url
-    page_title = "New Country"
+    page_title = "Add Country"
     permission_required = Perms.sys.ilb_admin
+    cancel_url = reverse_lazy("country:list")
+
+    def form_valid(self, form):
```
Comment (MattHolmes123) [inline web/domains/country/views.py:69]: missing types

**Review by MattHolmes123 — APPROVED** [review]
I've approved this but I think @mikeylockwood is better placed to give it a real review.

File: web/templates/web/domains/country/add.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+{% block sidebar %}
+  {% include "web/domains/country/sidebar.html" %}
+{% endblock %}
+
+{% block fields %}
+  {% set tooltip %}
+    {% include "web/domains/country/groups-tooltip.html" %}
+  {% endset %}
+  {{fields.field(form.name, tooltip=tooltip)}}
```
Comment (mikeylockwood) [inline web/templates/web/domains/country/add.html:12]: Minor but prefer whitespace within template tags
`{{ fields.field(form.name, tooltip=tooltip) }}`

File: web/templates/web/domains/country/edit.html [hunk 5]
```diff
@@ -14,6 +14,9 @@
 
   {{fields.field(form.name, tooltip=tooltip)}}
   {{fields.field(form.type)}}
+  {{fields.field(form.overseas_region)}}
+  {{fields.field(form.is_active)}}
```
Comment (mikeylockwood) [inline web/templates/web/domains/country/edit.html:18]: Template tag whitespace for this entire file

File: data_migration/management/commands/config/data_counts.py [hunk 6]
```diff
@@ -49,6 +49,23 @@
...
+        model=web.Country,
+        filter_params={"overseas_region__isnull": True, "type": web.Country.SOVEREIGN_TERRITORY},
+    ),
+    CheckCount(
+        name="Countries",
+        expected_count=269,
+        model=web.Country,
+    ),
+    CheckCount(
+        name="Countries that are active",
```
Comment (mikeylockwood) [inline data_migration/management/commands/config/data_counts.py:64]: Inactive countries are ones created by add_v2_reference_data and not the data migration?

**Review by mikeylockwood — APPROVED** [review]

File: data_migration/management/commands/config/data_counts.py (hunk 6 above)
Comment (marcuspp) [reply data_migration/management/commands/config/data_counts.py:64]: Correct. So check for total count and a check for the data migration part that you are left with the correct amount of active countries.

**Pull request merged**